cli/src/pcluster3_config_converter/pcluster3_config_converter.py

```diff
@@ -0,0 +1,1000 @@
+            self.convert_single_field(section_label, "enable", cloudwatch, "Enabled", "getboolean")
+            self.convert_single_field(section_label, "retention_days", cloudwatch, "RetentionInDays", "getint")
+
+        dashboards = dict()
+        dashboard_cloudwatch = dict()
+        dashboard_settings = self.cluster_config_get("dashboard_settings")
+        if dashboard_settings:
+            section_label = f"dashboard {dashboard_settings.strip()}"
+            self.convert_single_field(section_label, "enable", dashboard_cloudwatch, "Enabled", "getboolean")
+
+        _add_if(logs, "CloudWatch", cloudwatch)
+        _add_if(monitoring, "Logs", logs)
+        _add_if(dashboards, "CloudWatch", dashboard_cloudwatch)
+        _add_if(monitoring, "Dashboards", dashboards)
+        _add_if(self.pcluster3_configuration, section_name, monitoring)
+
+    def convert_headnode(self, section_name):
+        """Convert HeadNode section."""
+        headnode = dict()
+        self.convert_instance_type(headnode)
+        self.convert_networking(headnode)
+        self.convert_disable_hyperthreading(headnode)
+        self.convert_ssh(headnode)
+        self.convert_local_storage(headnode, "master_root_volume_size")
+        self.convert_dcv(headnode)
+        self.convert_custom_action(headnode)
+        self.covert_headnode_iam(headnode)
+        self.convert_imds(headnode)
+        _add_if(self.pcluster3_configuration, section_name, headnode)
+
+    def convert_single_field(
+        self, pcluster2_model, pcluster2_attribute, pcluster3_model, pcluster3_attribute, function_name=None
+    ):
+        """Convert a single field from config_parser object to parameter in pcluster3 model."""
+        try:
+            function_map = {
+                "getboolean": self.config_parser.getboolean,
+                "getint": self.config_parser.getint,
+                "getfloat": self.config_parser.getfloat,
+            }
+            function = function_map.get(function_name, self.config_parser.get)
+            attribute = function(pcluster2_model, pcluster2_attribute, fallback=None)
+            if attribute is not None:
+                pcluster3_model[pcluster3_attribute] = attribute
+        except Exception as e:
+            _error(f"Wrong type for {pcluster2_attribute} in {pcluster2_model} section: {e}")
+
+    def write_configuration_file(self):
+        """Write converted yaml to the output file."""
+        if not os.path.isfile(self.output_file):
+            try:
+                config_folder = os.path.dirname(self.output_file) or "."
+                os.makedirs(config_folder)
+            except OSError as e:
+                if e.errno != errno.EEXIST:
+                    raise
+
+            # Fix permissions
+            with open(self.output_file, "a", encoding="utf-8"):
+                os.chmod(self.output_file, stat.S_IRUSR | stat.S_IWUSR)
+
+        # Write configuration to disk
+        if self.output_file:
```

Comment:
Curious what this guard is / why it's here? If we don't have an output file it seems like there's some other error that should be caught elsewhere / earlier (?)

Otherwise, it seems like it's worth returning earlier in this function when it isn't specified (e.g. basically as a pre-condition)

Reply:
Fixed. I miss to change this part and forgot about adding it to stdout.
Now fixed the function to add if there's no self.out_outfile, write the result to stdout
```
    def write_configuration_file(self):
        """Write converted yaml to the output file."""
        if self.output_file:
            if os.path.isfile(self.output_file):
                _error(f"File {self.output_file} already exists, please select another output file.")
            else:
                try:
                    config_folder = os.path.dirname(self.output_file) or "."
                    os.makedirs(config_folder)
                except OSError as e:
                    if e.errno != errno.EEXIST:
                        raise
            # Fix permissions
            with open(self.output_file, "a", encoding="utf-8"):
                os.chmod(self.output_file, stat.S_IRUSR | stat.S_IWUSR)

            # Write configuration to disk
            with open(self.output_file, "w", encoding="utf-8") as config_file:
                config_file.write(self.comments)
                yaml.dump(
                    self.pcluster3_configuration,
                    config_file,
                    sort_keys=True,
                    Dumper=ConfigDumper,
                    default_flow_style=False,
                )
            print(f"Configuration file written to {self.output_file}")
        else:
            sys.stdout.write(self.comments)
            yaml.dump(
                self.pcluster3_configuration,
                sys.stdout,
                sort_keys=True,
                Dumper=ConfigDumper,
                default_flow_style=False,
            )
            sys.stdout.flush()
```